src/stepfunctions/inputs/placeholders.py

```diff
@@ -291,3 +291,26 @@ def _create_variable(self, name, parent, type=None):
             return StepInput(name=name, parent=parent, type=type)
         else:
             return StepInput(name=name, parent=parent)
+
+
+class StepResult(Placeholder):
+
+    """
+        Top-level class for step result placeholders.
+    """
+
+    def __init__(self, schema=None, **kwargs):
+        super(StepResult, self).__init__(schema, **kwargs)
+        self.json_str_template = '${}'
+
+    def _create_variable(self, name, parent, type=None):
+        """
+            Creates a placeholder variable for Step Result.
+            A placeholder variable can only be created if the collection is not immutable due to a pre-specified schema.
```

Comment:
```suggestion
            A placeholder variable can only be created if the collection is mutable due to a pre-specified schema.
```

rationale: just easier to understand imo

Reply:
"immutable" was used to point to the `immutable` object attribute
In that sense. would it be acceptable to keep it as is?

Reply:
i'd still suggest getting rid of the double negative. it's a nit comment. It also helps for the error message to suggest how to correct the validation error.

Reply:
Double negation was removed in latest commit